python/whylogs/api/writer/whylabs.py

```diff
@@ -558,8 +554,13 @@ def write(self, file: Writable, **kwargs: Any) -> Tuple[bool, str]:
             return self.write_feature_weights(file, **kwargs)
         elif isinstance(file, EstimationResult):
             return self.write_estimation_result(file, **kwargs)
-        elif isinstance(file, SegmentedResultSet) and self._reference_profile_name is not None:
-            return self._write_segmented_reference_result_set(file, **kwargs)
+        elif isinstance(file, ResultSet):
+            if isinstance(file, SegmentedResultSet) and self._reference_profile_name is not None:
+                return self._write_segmented_reference_result_set(file, **kwargs)
+                # TODO: why does segmented reference profile need special handling, but
```

Comment:
Its because segmented reference profile has a special case API server side, so that it can treat N files as a single profile (where N is the number of individual segments), and returns back a single reference profile-id for all the segments uploaded in this single call.

Non-reference segmented profiles upload each segment as a separate file with a separate upload by design, there is no notion of completeness and the segments are not referenceable as a group by profile-id, you can add a new segment later, and the combined merge of all segments has no profile-id. These are platform specific considerations for being able to define a segmented reference to use as a baseline for monitoring.